pkg/workspace/templates.go

```diff
@@ -145,17 +145,26 @@ func cleanupLegacyTemplateDir() error {
 	return nil
 }
 
-// IsTemplateURL returns true if templateNameOrURL starts with "https://".
-func IsTemplateURL(templateNameOrURL string) bool {
-	return strings.HasPrefix(templateNameOrURL, "https://")
+// IsTemplateURL returns true if templateNamePathOrURL starts with "https://".
+func IsTemplateURL(templateNamePathOrURL string) bool {
+	return strings.HasPrefix(templateNamePathOrURL, "https://")
 }
 
-// RetrieveTemplates retrieves a "template repository" based on the specified name or URL.
-func RetrieveTemplates(templateNameOrURL string, offline bool) (TemplateRepository, error) {
-	if IsTemplateURL(templateNameOrURL) {
-		return retrieveURLTemplates(templateNameOrURL, offline)
+// isTemplateFileOrDirectory returns true if templateNamePathOrURL is the name of a valid file or directory.
+func isTemplateFileOrDirectory(templateNamePathOrURL string) bool {
+	_, err := os.Stat(templateNamePathOrURL)
```

Comment:
This does mean that if I have a dir named `typescript` and I run `pulumi new typescript -f`, it'll now use that local directory over fetching the _actual_ template from GitHub. But getting into that situation would require the `-f` (`--force`) flag (since the current directory wouldn't be empty if it includes a `typescript` dir) and actually having a local dir named the same as one of the templates. That seems sufficiently edge case that I think this is "OK".

I was wondering if we should require these to be specified as `file://` instead to avoid that, but that makes it more annoying to specify local templates, and I don't think `file://` works well with relative paths.

Reply:
> I was wondering if we should require these to be specified as file:// instead to avoid that, but that makes it more annoying to specify local templates, and I don't think file:// works well with relative paths.

I suspect we could make something like `file://./a/relative/path` behave in a reasonable fashion (the first component after the protocol is a `.` for the CWD)

Reply:
I thought about this but found `file://` paths to be a bit ugly. Let me know if you think it's still worth it, and I'll try to test different path cases.

Reply:
> I suspect we could make something like `file://./a/relative/path` behave in a reasonable fashion (the first component after the protocol is a `.` for the CWD)

We could, but it'd go against the file URI scheme as [spec'd](https://tools.ietf.org/html/rfc8089#section-2) where the part after the `//` should be the "host":

>   The "host" is the fully qualified domain name of the system on which
   the file is accessible.  This allows a client on another system to
   know that it cannot access the file system, or perhaps that it needs
   to use some other local mechanism to access the file.
> 
>   As a special case, the "file-auth" rule can match the string
   "localhost" that is interpreted as "the machine from which the URI is
   being interpreted," exactly as if no authority were present.  Some
   current usages of the scheme incorrectly interpret all values in the
   authority of a file URI, including "localhost", as non-local.  Yet
   others interpret any value as local, even if the "host" does not
   resolve to the local machine.  To maximize compatibility with
   previous specifications, users MAY choose to include an "auth-path"
   with no "file-auth" when creating a URI.

I'm OK with supporting a path here and not requiring the use of `file://`.